src/js/ProviderManager.js

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+/*globals async, DirectoryLinksProvider, Links, PlacesProvider, gUserDatabase, gPinnedLinks, gBlockedLinks*/
+
+/**
+ * Singleton that handles link providers.
+ */
+
+ "use strict";
+
+const ProviderManager = {
+  init: async(function*(gMockFetch) {
```

Comment:
@marcoscaceres - either of those sound good to me. It sounds like you know exactly how these would work. I wouldn't mind if you added a commit with those changes to this branch :)